### PR TITLE
Remove wasm-support-plugin and deal with the code

### PR DESCRIPTION
**Pull request by andreaTP**
Alternative ( to #514 ) fix for #513 

I think this is simpler and, at the end of the day, more easily maintainable (fun fact: I have seen multiple ppl stumbling on the `.tsv` as IntelliJ was not respecting the tabs :sweat_smile: ).

### CODE REVIEW

**Comment by jmillikin** [comment]
(author of #513 / #514 here)

I don't have a strong preference either way -- if this PR gets merged I'll close mine.

If you expect to add more instructions over time (for example when adding support for SIMD or multi-memory) then keeping the table might be better. Java is not generally considered a terse language, so having a bit of codegen can be useful for making the programmer's intent clear. 

> fun fact: I have seen multiple ppl stumbling on the `.tsv` as IntelliJ was not respecting the tabs 😅 

I was definitely surprised to see TSV used here. Similar tables I've seen in other projects treat tabs and spaces as just plain whitespace, which tends to survive editor round-trips better. For an example see the Linux kernel's [`arch/arm/tools/syscall.tbl`](https://github.com/torvalds/linux/blob/v6.10/arch/arm/tools/syscall.tbl). I've also seen people use structured formats such as lines of JSON, which can be nice when items are heterogenous.

Regardless of which approach is used, thank you for taking the time. I'll answer your questions in the other PR in case that helps with the decision.

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+
+    private static final Map<OpCode, WasmEncoding[]> signature = new HashMap<>();
+
+    public static WasmEncoding[] getSignature(OpCode o) {
+        return signature.get(o);
+    }
+
+    static {
+        for (OpCode e : OpCode.values()) {
+            byOpCode.put(e.opcode(), e);
```
Comment (electrum) [inline wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java:437]: Might want a validation here
```java
if (byOpCode.put(e.opcode(), e) != null) {
    throw new AssertionError("duplicate opcode: " + e.opcode());
}

**Review by electrum — APPROVED** [review]
I like getting rid of the code generation for this. Adding new instructions here is very minor compared to the rest of the implementation.

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+
+    private static final Map<Integer, OpCode> byOpCode = new HashMap<>();
+
+    public static OpCode byOpCode(int opcode) {
+        return byOpCode.get(opcode);
+    }
+
+    private static final Map<OpCode, WasmEncoding[]> signature = new HashMap<>();
+
+    public static WasmEncoding[] getSignature(OpCode o) {
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java]: done

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java (hunk 1 above)
Comment (andreaTP) [reply wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java]: Moved the Map to be an array, I see the value of the check, but is, ultimately, only for us developers of the engine.
I left it out for now.

**Comment by andreaTP** [comment]
@electrum moved to a fully hand rolled implementation, let me know what you think about this!

**Review by electrum — APPROVED** [review]
Looks great!

I don't know if an array is needed here. With 8 byte references, each array is 512kB, so this adds a 1MB memory overhead. Does this provide a measurable improvement when parsing WASM files?

If there is no improvement and the memory overhead is a problem for someone, we can switch back to a hash map later.

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java (hunk 1 above)
Comment (electrum) [reply wasm/src/main/java/com/dylibso/chicory/wasm/types/OpCode.java]: Agreed, and you would find the error immediately anyway, so it's not needed.

File: wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java [hunk 3]
```diff
@@ -906,7 +906,7 @@ private static Instruction parseInstruction(ByteBuffer buffer) {
         }
 
         // System.out.println("b: " + b + " op: " + op);
-        var signature = OpCode.getSignature(op);
+        var signature = OpCode.signature(op);
         // TODO: Encode this in instructions.tsv ?
```
Comment (electrum) [inline wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java:910]: Remove outdated comment

Comment (andreaTP) [reply wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java]: done

**Pull request merged**